src/coll/mod.rs

```diff
@@ -318,6 +469,39 @@ where
         cursor.next().await.transpose()
     }
 
+    /// Finds a single document in the collection matching `filter` using the provided
+    /// `ClientSession`.
+    pub async fn find_one_with_session(
+        &self,
+        filter: impl Into<Option<Document>>,
+        options: impl Into<Option<FindOneOptions>>,
+        session: &mut ClientSession,
+    ) -> Result<Option<T>> {
+        let mut options: FindOptions = options
+            .into()
+            .map(Into::into)
+            .unwrap_or_else(Default::default);
+        options.limit = Some(-1);
```

Comment:
I think this limit line actually ought to be in the `From<FindOneOptions>` implementation for `FindOptions`

Reply:
Ah yep, I was probably to blame for the original implementation then :slightly_smiling_face: 

BTW, this assignment can be removed now that the `From` implementation takes care of it.